src/openeo_gfmap/manager/job_manager.py

```diff
@@ -16,21 +21,53 @@
 from openeo_gfmap.stac import constants
 
 # Lock to use when writing to the STAC collection
-_stac_lock = threading.Lock()
+_stac_lock = Lock()
+
+
+def retry_on_exception(max_retries, delay=30):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            latest_exception = None
+            for _ in range(max_retries):
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    time.sleep(delay)
+                    latest_exception = e
+            raise latest_exception
+
+        return wrapper
+
+    return decorator
```

Comment:
could you include  a docstring and the expected type for each input/output?

Reply:
Another question is on the preset delay, for what do you aim to build resilience? HTTP errors or general cdse issues? Perhaps it could be better to set it as a standard to a longer duration or move said job back to the back of the queue?

Reply:
Internal errors from OpenEO in the CDSE backend indeed. Moving to the back the queue only isn't really a solution as they might in many case be picked up again instantly. I will increase the delay to 3 minutes with `time.sleep(...)`, which allows the executor to pick up other tasks in the meantime.